docs/explanations/flyscanning.md

```diff
@@ -0,0 +1,86 @@
+
+* Readback trigger: Same as for collection windows readback triggers.
+* Time-based trigger: Hardware triggers sent out at regularly spaced intervals in time.
+* Internal device trigger: The detector itself generates internal triggers based on its configuration.
+
+## Examples
+
+With all this in mind, let us consider typical flyscanning setups given a motor and a detector.
+
+### Software collection window internal triggered flyscan
+
+The simplest form of flyscanning uses software synchronisation for opening a collection window, and letting a detector trigger itself internally.
+We effectively tell the devices to "go" at the same time, relying on timestamps to match up the data. The strategy may involve
+
+* Calculating the motor run-up distance and move to the start, setting up the detector to take the right number of frames,
+* Starting the motor moving to its end point and waiting for the amount of time it takes to accelerate
+* Starting the detector
+* Waiting until the detector has finished taking frames and the motor has stopped moving
+
+This process would produce two data stream: a "monitor" stream of timestamped motor readback positions at the control system update rate;
+and a "primary" stream of detector frames with embedded timestamps when they were received by the control point.
+
+### Software collection window time triggered flyscan
+
+If we introduce a PandA box into the system we can trigger the collection window as above, but let the PandA produce time-based triggered for the detector.
+The strategy is similar:
+
+* Calculate motor run-up distance and move to the start, setting the detector into triggered mode and PandA to send the right number of triggers
+* Start the motor moving to its end point and wait for the amount of time it takes to accelerate
+* Start the PandA sending triggers
+* Wait until the PandA, detector and motor have finished
+
+The data streams produced are a "monitor" stream of timestamped motor readback positions at the control system update rate;
+and a "primary" stream of PandA captured timestamps and detector frames. If the motor happens to be plugged into the PandA
+then we can dispense with the monitor stream as the motor positions can be captured and produced by the PandA.
+
+### Readback collection window readback triggered flyscan
+
+If the motor positions are tracked in the PandA we can use a PCOMP block to trigger both collection windows and frames within.
+This gives the advantage that our detector will trigger at the precise motor positions we expect.
+The strategy and data streams are as in the previous example.
+
+### Bonus: setpoint collection window setpoint triggered hardware stepscan
+
+It is worth mentioning that the Flyer protocol is also used in hardware-triggered step scanning. Though the collection occurs while the scanned axis is stationary,
+the devices are prepared, kicked off, and completed in a similar way to flyscans.
```

Comment:
Should we enumerate the other 8-ish examples I'm aware of in this document?

Reply:
I will create a new issue to cover this, as discussed